components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -429,24 +439,28 @@ void JsonRpcService::UpdateIsEip1559(const std::string& chain_id,
   }
 }
 
-void JsonRpcService::FireNetworkChanged() {
+void JsonRpcService::FireNetworkChanged(mojom::CoinType coin) {
   for (const auto& observer : observers_) {
-    observer->ChainChangedEvent(GetChainId());
+    observer->ChainChangedEvent(GetChainId(coin), coin);
   }
 }
 
-std::string JsonRpcService::GetChainId() const {
-  return chain_id_;
+std::string JsonRpcService::GetChainId(mojom::CoinType coin) const {
+  if (!chain_ids_.contains(coin))
+    return std::string();
+  return chain_ids_.at(coin);
```

Comment:
nit: How about `return chain_ids_.contains(coin) ? chain_ids_[coin] : "";` ?

Reply:
fixed in https://github.com/brave/brave-core/pull/12461/commits/cea463abc30fc02b69d035171fd3525822495657